### ISSUE TITLE
az vm list not working

### ISSUE THREAD

**Issue by motorbreth** [Azure/azure-cli#24187]
## Describe the bug

**Command Name**
`az vm list`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'ComputeManagementClientConfiguration' object has no attribute 'api_version'
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/vm/custom.py", line 1381, in list_vm
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/core/paging.py", line 129, in __next__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/core/paging.py", line 76, in __next__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/mgmt/compute/v2022_08_01/operations/_virtual_machines_operations.py", line 2254, in get_next
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/mgmt/compute/v2022_08_01/operations/_virtual_machines_operations.py", line 2239, in prepare_request
AttributeError: 'ComputeManagementClientConfiguration' object has no attribute 'api_version'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az vm list`

## Environment Summary
```
Windows-10-10.0.22000-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.41.0
```

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

**Comment by zhoxing-ms** [comment]
This problem was introduced by Python SDK https://github.com/Azure/azure-sdk-for-python/issues/26506. I see they had submitted PR https://github.com/Azure/autorest.python/issues/1485 to fix the bug in package version `28.0.1`.
But I found that there are other problems with version `28.0.1` https://github.com/Azure/azure-sdk-for-python/issues/26604, which are also being solved

@msyyc May I ask when will the new package which solve all problems be released? Could you inform us at that time?

**Comment by zhoxing-ms** [comment]
Please help upgrade the SDK version to [azure-mgmt-compute/29.0.0](https://pypi.org/project/azure-mgmt-compute/29.0.0/) to solve the problem. 

@motorbreth @techny210 We will release a new CLI version in `11-01` to fix this bug, thanks for your feedback~

**Comment by chester-leung** [comment]
Hi, I found the same issue. I did not install any Azure Python SDK on the machine, only the Azure CLI. What should I do to workaround this bug?

**Comment by zhoxing-ms** [comment]
@chester-leung In fact, the Python and Python SDK packages are embedded in Azure CLI package. The temporary workaround is to downgrade the CLI version. In 2 weeks **(11-01)**, we will release a new official version to fix it~

**Comment by chester-leung** [comment]
@zhoxing-ms thanks for your reply. Which Azure CLI version will work? Can you share some specific instructions on how to downgrade to a working version? Thank you.

**Comment by zhoxing-ms** [comment]
@chester-leung We will solve this bug in the next CLI version `2.42.0`. At present, the way to downgrade the CLI version is to uninstall the current version of CLI first, and then install the old version of CLI. [doc link](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)

**Comment by EdgeSN** [comment]
> 

is the fix still planned for 11-01 ?

**Comment by yonzhan** [comment]
Yes, it will be released on 11.1
